src/lib/hooks/useNavigationStateHandler.js

```diff
@@ -4,13 +4,33 @@ import { useCurriedSetters } from '../undux/SimpleStore.js'
 import { useDialog } from '../undux/utils/dialog'
 import { fireEventFromNavigation } from '../analytics/analytics'
 
+const checkSameRoute = (prevNav, nav, prevIndexes = [], indexes = []) => {
+  const prevIndex = prevNav.index
+  const index = nav.index
+
+  if (prevIndex >= 0 && index >= 0) {
+    prevIndexes.push(prevIndex)
+    indexes.push(index)
+  } else {
+    return prevIndexes.join('') === indexes.join('')
+  }
+
+  return checkSameRoute(prevNav.routes[prevIndex], nav.routes[index], prevIndexes, indexes)
+}
+
 export default (options = {}) => {
   const [, hideDialog] = useDialog()
   const [setCurrentFeed, setSideMenu] = useCurriedSetters(['currentFeed', 'sidemenu'])
   const { resetFeed = true, resetMenu = true, resetPopups = true, fireEvent = true, onChange = noop } = options
 
   return useCallback(
     (prevNav, nav, action) => {
+      // do not clean the dialog state if the route is not changed
+      const isSameRoute = checkSameRoute(prevNav, nav)
+      if (isSameRoute) {
```

Comment:
```suggestion
      const isSameRoute = getRoutePath(prevNav) === getRoutePath(nav)
     // also you could import getRouteName from stackNavigation and try:
     // const isSameRoute = getRouteName(prevNav) === action.routeName
     // This will be faster but not sure if will work correctly, pls check
     
     // let's pass isSameRoute flag to possible custom callback
     onChange(prevNav, nav, action, isSameRoute)
      
      if (isSameRoute) {
```

Reply:
and pls remove onChange call from the line 50

Reply:
I tested
```
const isSameRoute = getRoutePath(prevNav) === getRoutePath(nav)
```
it is working as required!